src/rust/engine/src/nodes.rs

```diff
@@ -334,8 +334,11 @@ impl ExecuteProcess {
   ) -> Result<Process, StoreError> {
     let env = externs::getattr_from_str_frozendict(value, "env");
     let working_directory = match externs::getattr_as_optional_string(value, "working_directory") {
-      None => None,
-      Some(dir) => Some(RelativePath::new(dir)?),
+      Ok(dir) => match dir {
+        Some(d) => Some(RelativePath::new(d)?),
+        None => None,
+      },
+      Err(_) => None,
```

Comment:
All callers of `getattr_as_optional_string` should use `.map_err(..)?`, for the same reason: there is also an `unwrap` added below that should be replaced.